### PR TITLE
GH-35875: [R] Update Readme

### LINKED ISSUE #35875: [R][Documentation] Improve text in top-level package README
### Describe the enhancement requested

As discussed on the 2023/06/01 Arrow R package development sync call, a reworking of the [top-level README](https://arrow.apache.org/docs/r/) might make it easier for potential users of the package to understand what they can do and more likely for them to become users. While the current content gives a good overview, there are some areas it could be improved:

1. Move the discussion of Apache Arrow itself lower down
2. Describe a smaller and denser list of features, some could be combined, some could be moved or removed. Maybe five or so
3. Link each feature to more detail about how to use that feature

The [sparklyr](https://spark.rstudio.com) package website was brought up as an example and I think it's a good one. It lists a denser feature set and links to internal documentation for each. Their [guides](https://spark.rstudio.com/guides/) section was also brought up because of how it organizes guides.

Please comment if you have any ideas or suggestions.

cc @thisisnic @stephhazlitt @fmichonneau 

### Component(s)

Documentation, R

### PR DESCRIPTION
**Pull request by dgreiss**
### Rationale for this change

#35875  #35082 and #32895 make a number of recommendations to update the the Readme

### What changes are included in this PR?

Rewording and reorganizing the Readme and sidebar.

### Are these changes tested?
n/a

### Are there any user-facing changes?

Yes 

* Closes: #35875

### CODE REVIEW

**Comment by amoeba** [comment]
@github-actions crossbow submit preview-docs

File: r/README.md [hunk 1]
```diff
@@ -1,114 +1,96 @@
...
--   Toolkit for building connectors to other applications
-    and services that use Arrow
+To learn more about the Apache Arrow project, see the parent documentation of the [Arrow Project](https://arrow.apache.org/). The Arrow project provides functionality for a wide range of data analysis tasks to store, process and move data fast. See the [read/write article](articles/read_write.html) to learn about reading and writing data files, [data wrangling](article/data_wrangling.html) to learn how to use dplyr syntax with arrow objects, and the [function documentation](reference/acero.html) for a full list of supported functions within dplyr queries.
 
 ## Installation
 
-Most R users will probably want to install the latest release of arrow 
-from CRAN:
+The latest release of arrow can be installed from CRAN. In most cases installing the latest release should work without requiring any additional system dependencies, especially if you are using
+Windows or a Mac.
```
Comment (amoeba) [inline r/README.md:20]:
```suggestion
Windows or macOS.
```

**Review by amoeba — COMMENTED** [review]
Hi @dgreiss, thanks for the follow-up PR. A few high-level points:

- Is it possible at this point to update your PR without all the whitespace changes? It takes extra work to figure out what text actually changed
- Could the description of Apache Arrow be kept but just put lower down? I don't feel like the README would be too long with it included

**Comment by thisisnic** [comment]
Thanks for updating this PR @dgreiss!  

@amoeba - as this has been so long-running and we have other folks interested in developing the docs further, I suggest we push your suggested changes to this PR ourselves and then merge it.

**Comment by amoeba** [comment]
Sounds good @thisisnic, I'll do that today.

**Comment by amoeba** [comment]
@thisisnic are you able to edit the PR? I don't think I can. Here's my patch that addresses my comments. This is also available on https://github.com/amoeba/arrow/tree/gh-35875-dgreiss-update-r-readme.

```patch
diff --git a/r/README.md b/r/README.md
index 8b6a050055..1f953a8e6f 100644
--- a/r/README.md
+++ b/r/README.md
@@ -12,7 +12,7 @@
 
 The R `{arrow}` package provides access to many of the features of the [Apache Arrow C++ library](https://arrow.apache.org/docs/cpp/index.html) for R users. The goal of arrow is to provide an Arrow C++ backend to `{dplyr}`, and access to the Arrow C++ library through familiar base R and tidyverse functions, or `{R6}` classes.
 
-To learn more about the Apache Arrow project, see the parent documentation of the [Arrow Project](https://arrow.apache.org/). The Arrow project provides functionality for a wide range of data analysis tasks to store, process and move data fast. See the [read/write article](articles/read_write.html) to learn about reading and writing data files, [data wrangling](article/data_wrangling.html) to learn how to use dplyr syntax with arrow objects, and the [function documentation](reference/acero.html) for a full list of supported functions within dplyr queries.
+To learn more about the Apache Arrow project, see the parent documentation of the [Arrow Project](https://arrow.apache.org/). The Arrow project provides functionality for a wide range of data analysis tasks to store, process and move data fast. See the [read/write article](articles/read_write.html) to learn about reading and writing data files, [data wrangling](articles/data_wrangling.html) to learn how to use dplyr syntax with arrow objects, and the [function documentation](reference/acero.html) for a full list of supported functions within dplyr queries.
 
 ## Installation
 
@@ -65,6 +65,18 @@ Additional features include:
 - Fine control over column types to work seamlessly with databases and data warehouses
 - Toolkit for building connectors to other applications and services that use Arrow
 
+## What is Apache Arrow?
+
+Apache Arrow is a cross-language development platform for in-memory and
+larger-than-memory data. It specifies a standardized language-independent
+columnar memory format for flat and hierarchical data, organized for efficient
+analytic operations on modern hardware. It also provides computational libraries
+and zero-copy streaming, messaging, and interprocess communication.
+
+This package exposes an interface to the Arrow C++ library, enabling access to
+many of its features in R. It provides low-level access to the Arrow C++ library
+API and higher-level access through a dplyr backend and familiar R functions.
+
 ## Arrow resources
 
 There are a few additional resources that you may find useful for getting started with arrow:
@@ -85,7 +97,10 @@ the [Apache Arrow Community](https://arrow.apache.org/community/) page.
 If you encounter a bug, please file an issue with a minimal reproducible
 example on [GitHub issues](https://github.com/apache/arrow/issues).
 Log in to your GitHub account, click on **New issue** and select the type of
-@@ -104,11 +92,8 @@ features\*\* section of the [Contributing to Apache
+issue you want to create. Add a meaningful title prefixed with **`[R]`**
+followed by a space, the issue summary and select component **R** from the
+dropdown list. For more information, see the **Report bugs and propose
+features** section of the [Contributing to Apache
 Arrow](https://arrow.apache.org/docs/developers/#contributing) page
 in the Arrow developer documentation.
 ```

File: r/README.md [hunk 2]
```diff
@@ -1,114 +1,96 @@
...
-CentOS 7. For Windows users it means you need to be running an R version of 
-4.0 or later. On CentOS 7, it means you need to install a newer compiler 
-than the default system compiler gcc 4.8. See the [installation details article](https://arrow.apache.org/docs/r/articles/developers/install_details.html) for guidance. Note that 
-this does not affect users who are installing a binary version of the package.
+Additional features include:
+
+- Manipulate and analyze Arrow data with dplyr verbs
+- Zero-copy data sharing between R and Python
+- Fine control over column types to work seamlessly with databases and data warehouses
+- Toolkit for building connectors to other applications and services that use Arrow
```
Comment (thisisnic) [inline r/README.md:66]:
```suggestion
- Toolkit for building connectors to other applications and services that use Arrow

## What is Apache Arrow?

Apache Arrow is a cross-language development platform for in-memory and
larger-than-memory data. It specifies a standardized language-independent
columnar memory format for flat and hierarchical data, organized for efficient
analytic operations on modern hardware. It also provides computational libraries
and zero-copy streaming, messaging, and interprocess communication.

This package exposes an interface to the Arrow C++ library, enabling access to
many of its features in R. It provides low-level access to the Arrow C++ library
API and higher-level access through a dplyr backend and familiar R functions.

```

File: r/README.md [hunk 3]
```diff
@@ -1,114 +1,96 @@
...
+the [Apache Arrow Community](https://arrow.apache.org/community/) page.
+
 If you encounter a bug, please file an issue with a minimal reproducible
 example on [GitHub issues](https://github.com/apache/arrow/issues).
 Log in to your GitHub account, click on **New issue** and select the type of
-issue you want to create. Add a meaningful title prefixed with **`[R]`**
-followed by a space, the issue summary and select component **R** from the
-dropdown list. For more information, see the **Report bugs and propose
-features** section of the [Contributing to Apache
+@@ -104,11 +92,8 @@ features\*\* section of the [Contributing to Apache
```
Comment (thisisnic) [inline r/README.md:88]:
```suggestion
issue you want to create. Add a meaningful title prefixed with **`[R]`**
followed by a space, the issue summary and select component **R** from the
dropdown list. For more information, see the **Report bugs and propose
features** section of the [Contributing to Apache
```

**Comment by thisisnic** [comment]
Thanks @amoeba!

FYI another alternative route that I take when I have had a long day and don't want to have to remind myself how to push to someone else's branch is also to just apply the changes as "suggestions", and then if you're a committer, you should be able to accept those suggestions on the PR.

**Comment by thisisnic** [comment]
This is ready for merging once CI passes.  Thanks very much @dgreiss, the readme is looking a lot more straightforward now! :)

**Review by thisisnic — APPROVED** [review]

**Comment by amoeba** [comment]
Ah, that's a nice trick, thanks @thisisnic. Still working on getting that commit bit :)

**Pull request merged**